modules/README.md

```diff
@@ -141,9 +145,123 @@ interface ISafeProtocol712SignatureValidator {
         bytes32 domainSeparator,
         bytes32 typeHash,
         bytes calldata encodeData,
-        bytes calldata payload
+        bytes calldata signature
+        bytes calldata additionalData
     ) external view returns (bytes4 magic);
 }
 ```
 
+- Signature Validator Manager
+
+
+```solidity
+interface ISafeProtocolSignatureValidatorManager {
+    /**
+     * @param dataHash Hash of the data that is signed
+
+     * @return bytes4 Value returned by the Signature Validator Contract
+     */
+    function isSignatureValid(bytes32 dataHash, bytes data) return (bytes4 magicValue);
+
+    /**
+     * @notice A view function that the Manager will call when an account has enabled this contract as a function handler in the Manager for function isSignatureValid(bytes32,bytes)
+     * @param account Address of the account whose signature validator is to be used
+     * @param sender Address requesting signature validation
+     * @param data Calldata containing the function selector, signature hash, domain separator, type hash, encoded data and payload forwarded by the Manager
+     *              Arbitrary data containing the following layout:
+     *              Layout of the data:
+     *                  0x00 to 0x04: 4 bytes function selector
+     *                  0x04 to 0x36: dataHash
+     *                  0x36 to 0x68: domainSeparator
+     *                  0x68 to 0x9a: typeHash
+     *                  0x9a to 0xcc: encodeData length
+     *                  0xcc to <0xcc + encodeData length>: encodeData
+     *                  <0xcc + encodeData length> to <0xcc + encodeData length> + 0x32 : signature length
```

Comment:
shouldn't this be `<0xcc + encodeData length + 0x20>`?

In most of these fields it seems that a decimal used even so it is a hex representation. A hash is 32 bytes wide, so 0x20 in hex encoding.

Reply:
ISafeProtocol712SignatureValidator